### PR TITLE
Adding a expect_status flag in xds interop client to get ready for

### PR DESCRIPTION
**Pull request by donnadionne**
timeout test.

---
This change is [Reviewable](https://reviewable.io/reviews/grpc/grpc/24996)

### CODE REVIEW

**Review by donnadionne — COMMENTED** [review]
I think this is all we need to add in the client; let me know if i Missed something.

File: test/cpp/interop/xds_interop_client.cc [hunk 1]
```diff
@@ -55,6 +55,8 @@ ABSL_FLAG(int32_t, stats_port, 50052,
 ABSL_FLAG(std::string, rpc, "UnaryCall",
           "a comma separated list of rpc methods.");
 ABSL_FLAG(std::string, metadata, "", "metadata to send with the RPC.");
+ABSL_FLAG(std::string, expect_status, "OK",
```
Comment (markdroth) [inline test/cpp/interop/xds_interop_client.cc:58]: I believe the absl flag API provides the ability to register a verifier function to be run when a value is specified for a flag.  We should use that to ensure at startup time that we are not given an invalid value here.

File: test/cpp/interop/xds_interop_client.cc [hunk 2]
```diff
@@ -345,6 +347,18 @@ class TestClient {
     std::unique_ptr<ClientAsyncResponseReader<SimpleResponse>>
         simple_response_reader;
   };
+  static bool RpcStatusCheckSuccess(AsyncClientCall* call) {
+    // Determine RPC success based on expected status.
+    if (absl::GetFlag(FLAGS_expect_status) == "OK" && call->status.ok()) {
```
Comment (markdroth) [inline test/cpp/interop/xds_interop_client.cc:352]: Please use `status_code_from_string()` here.

https://github.com/grpc/grpc/blob/64f04e4d2fad514c56bfd3a9145f6403c3697281/src/core/lib/channel/status_util.h#L31

Then you can just do something like this:

```
grpc_status_code code;
GPR_ASSERT(grpc_status_code_from_string(
    absl::GetFlag(FLAGS_expect_status).c_str(), &code));
return code == call->status.error_code();
```

**Review by donnadionne — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/grpc/grpc/24996)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @markdroth and @voidzcy)

___
*[test/cpp/interop/xds_interop_client.cc, line 58 at r1](https://reviewable.io/reviews/grpc/grpc/24996#-MOiDAGGLVmuekTH-hg5-r1-58:-MOiDC0hBd8FKRWg17nR:b61xraa) ([raw file](https://github.com/grpc/grpc/blob/abab1a399780b83f46b925b7bf4a79ad3a5453c8/test/cpp/interop/xds_interop_client.cc#L58)):*
<details><summary><i>Previously, markdroth (Mark D. Roth) wrote…</i></summary><blockquote>

I believe the absl flag API provides the ability to register a verifier function to be run when a value is specified for a flag.  We should use that to ensure at startup time that we are not given an invalid value here.
</blockquote></details>

fixed; register_validator is only available in python, OnUpdate is the closest in C++ (but it's not a public API yet).
___
*[test/cpp/interop/xds_interop_client.cc, line 352 at r1](https://reviewable.io/reviews/grpc/grpc/24996#-MOiDAGGLVmuekTH-hg5-r1-352:-MOiDQRO5N9IbyUi-eWf:b-896fix) ([raw file](https://github.com/grpc/grpc/blob/abab1a399780b83f46b925b7bf4a79ad3a5453c8/test/cpp/interop/xds_interop_client.cc#L352)):*
<details><summary><i>Previously, markdroth (Mark D. Roth) wrote…</i></summary><blockquote>

Please use `status_code_from_string()` here.

https://github.com/grpc/grpc/blob/64f04e4d2fad514c56bfd3a9145f6403c3697281/src/core/lib/channel/status_util.h#L31

Then you can just do something like this:

```
grpc_status_code code;
GPR_ASSERT(grpc_status_code_from_string(
    absl::GetFlag(FLAGS_expect_status).c_str(), &code));
return code == call->status.error_code();
```
</blockquote></details>

Done.

**Review by markdroth — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/grpc/grpc/24996)* status: all files reviewed, 2 unresolved discussions (waiting on @donnadionne and @voidzcy)

___
*[test/cpp/interop/xds_interop_client.cc, line 58 at r1](https://reviewable.io/reviews/grpc/grpc/24996#-MOiDAGGLVmuekTH-hg5-r1-58:-MOlEfPt8eYjfrna1ssl:bwlo127) ([raw file](https://github.com/grpc/grpc/blob/abab1a399780b83f46b925b7bf4a79ad3a5453c8/test/cpp/interop/xds_interop_client.cc#L58)):*
<details><summary><i>Previously, donnadionne wrote…</i></summary><blockquote>

fixed; register_validator is only available in python, OnUpdate is the closest in C++ (but it's not a public API yet).
</blockquote></details>

If `OnUpdate()` is not a public API, let's not use it.  Instead, we can just validate the value of the flag in `main()`.
___
*[test/cpp/interop/xds_interop_client.cc, line 64 at r2](https://reviewable.io/reviews/grpc/grpc/24996#-MOlEnKK3bbgO85uNQo3:-MOlEnKK3bbgO85uNQo4:b-i07l2l) ([raw file](https://github.com/grpc/grpc/blob/c4f8fa82391d909491136584b05d23d25503ae6d/test/cpp/interop/xds_interop_client.cc#L64)):*
> ```cpp
>     .OnUpdate([]() {
>       GPR_ASSERT(absl::c_linear_search(
>           std::vector<std::string>{"OK", "DEADLINE_EXCEEDED"},
> ```

We should accept any status code here, not just these two.  Please use `status_code_from_string()` here, just like we're doing below.

**Review by donnadionne — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/grpc/grpc/24996)* status: all files reviewed, 2 unresolved discussions (waiting on @markdroth and @voidzcy)

___
*[test/cpp/interop/xds_interop_client.cc, line 58 at r1](https://reviewable.io/reviews/grpc/grpc/24996#-MOiDAGGLVmuekTH-hg5-r1-58:-MOlXNi0-7ht2OpdVzx1:b-896fix) ([raw file](https://github.com/grpc/grpc/blob/abab1a399780b83f46b925b7bf4a79ad3a5453c8/test/cpp/interop/xds_interop_client.cc#L58)):*
<details><summary><i>Previously, markdroth (Mark D. Roth) wrote…</i></summary><blockquote>

If `OnUpdate()` is not a public API, let's not use it.  Instead, we can just validate the value of the flag in `main()`.
</blockquote></details>

Done.
___
*[test/cpp/interop/xds_interop_client.cc, line 64 at r2](https://reviewable.io/reviews/grpc/grpc/24996#-MOlEnKK3bbgO85uNQo3:-MOlXOCKDaVAlBBG5j2T:b-896fix) ([raw file](https://github.com/grpc/grpc/blob/c4f8fa82391d909491136584b05d23d25503ae6d/test/cpp/interop/xds_interop_client.cc#L64)):*
<details><summary><i>Previously, markdroth (Mark D. Roth) wrote…</i></summary><blockquote>

We should accept any status code here, not just these two.  Please use `status_code_from_string()` here, just like we're doing below.
</blockquote></details>

Done.

**Review by markdroth — APPROVED** [review]
> Reviewed 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/grpc/grpc/24996)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @voidzcy)

**Pull request merged**